src/libraries/Microsoft.PowerFx.Core/Parser/TexlParser.cs

```diff
@@ -202,1408 +205,1411 @@ private bool ParseUDF(List<UDF> udfs)
-        internal static ParseResult ParseScript(string script, CultureInfo loc = null, Flags flags = Flags.None)
+        }
+
+        // Parse the script
+        // Parsing strips out parens used to establish precedence, but these may be helpful to the
+        // caller, so precedenceTokens provide a list of stripped tokens.
+        internal static ParseResult ParseScript(string script, CultureInfo loc = null, Flags flags = Flags.None)
         {
-            return ParseScript(script, Features.None, loc, flags);
+            return ParseScript(script, Features.None, loc, flags);
         }
 
-        internal static ParseResult ParseScript(string script, Features features, CultureInfo loc = null, Flags flags = Flags.None)
-        {
-            Contracts.AssertValue(script);
-            Contracts.AssertValueOrNull(loc);
-
+        internal static ParseResult ParseScript(string script, Features features, CultureInfo loc = null, Flags flags = Flags.None)
+        {
+            Contracts.AssertValue(script);
+            Contracts.AssertValueOrNull(loc);
+
             var tokens = TokenizeScript(script, loc, flags);
-            var parser = new TexlParser(tokens, flags, features);
-            List<TexlError> errors = null;
-            var parsetree = parser.Parse(ref errors);
-
-            return new ParseResult(parsetree, errors, errors?.Any() ?? false, parser._comments, parser._before, parser._after, script, loc);
-        }
-
-        public static ParseFormulasResult ParseFormulasScript(string script, CultureInfo loc = null)
-        {
-            Contracts.AssertValue(script);
-            Contracts.AssertValueOrNull(loc);
-
-            var formulaTokens = TokenizeScript(script, loc, Flags.NamedFormulas);
-            var parser = new TexlParser(formulaTokens, Flags.NamedFormulas);
-
-            return parser.ParseFormulas(script);
-        }
-
-        private ParseFormulasResult ParseFormulas(string script)
-        {
-            var namedFormulas = new List<KeyValuePair<IdentToken, TexlNode>>();
-            ParseTrivia();
-
-            while (_curs.TokCur.Kind != TokKind.Eof)
-            {
-                // Verify identifier
-                var thisIdentifier = TokEat(TokKind.Ident);
-                if (thisIdentifier != null)
-                {
-                    ParseTrivia();
-
-                    // Verify "="
-                    var thisEq = TokEat(TokKind.Equ);
-                    if (thisEq != null)
-                    {
+            var parser = new TexlParser(tokens, flags, features);
+            List<TexlError> errors = null;
+            var parsetree = parser.Parse(ref errors);
+
+            return new ParseResult(parsetree, errors, errors?.Any() ?? false, parser._comments, parser._before, parser._after, script, loc);
+        }
+
+        public static ParseFormulasResult ParseFormulasScript(string script, CultureInfo loc = null)
```

Comment:
This needs a param to enable parsing number as float.
Also, in EnsureParsed in NamedFormulas.cs

Reply:
Added.  I added parser flags as a parameter, making it generic.